build-system/common/check-package-manager.js

```diff
@@ -33,7 +33,7 @@ const gulpHelpUrl =
 
 const yarnExecutable = 'npx yarn';
 const gulpExecutable = 'npx gulp';
-const pythonExecutable = 'python';
+const pythonExecutable = 'python2.7';
```

Comment:
I just realized this line will make this check for the default version of `python` ineffective:

https://github.com/ampproject/amphtml/blob/dd91119e7c7ddc5775f49a23cfcaa71c1b3358d7/build-system/common/check-package-manager.js#L312-L348

IIRC, the check was added because some of our dependencies internally invoke `python` without specifying the version (#23870). Not sure what the best course of action is, since your PR doesn't help those use cases.

Reply:
More context: https://github.com/ampproject/amphtml/issues/23801#issuecomment-519743182

Also, this line in [Nailgun v1.0](https://github.com/ampproject/amphtml/blob/master/third_party/nailgun/nailgun-runner), which is an integral part of our compilation toolchain, directly calls `python`:

https://github.com/ampproject/amphtml/blob/dd91119e7c7ddc5775f49a23cfcaa71c1b3358d7/third_party/nailgun/nailgun-runner#L1

Reply:
I think this PR handles this correctly (it certainly should be reviewed). The only two places in the build system where `python` was being invoked as a command (outside of `check-package-manager`) are in `build-system/tasks/validator.js`. These commands have been changed to `python2.7`. Since they both run `validator/build.py`, I didn't see the need to update the shebang in that python script.

Two other python scripts have had their shebang updated to make sure they use `python2.7`.

Please do scrutinize this PR for accuracy, but I think this is correct.

Reply:
SG. Let's have Daniel and the caching team look at this next week. Thanks for the attention to detail as always, Matt!